tests/integration/riofile_test.go

```diff
@@ -53,5 +55,43 @@ func riofileTests(t *testing.T, when spec.G, it spec.S) {
 			assert.Equal(t, "Hello World", routerFooV0.GetKubeEndpointResponse())
 			assert.Equal(t, "Hello World v3", routerFooV3.GetKubeEndpointResponse())
 		})
+
 	}, spec.Parallel())
+
+	when("A riofile with arbitrary k8s manifests", func() {
+		var riofile testutil.TestRiofile
+		const riofileName = "test-riofile-prune"
+		it.Before(func() {
+			riofile.Up(t, "riofile-with-K8s-manifests-before.yaml", riofileName)
+		})
+
+		it.After(func() {
+			riofile.Remove()
+		})
+
+		it("should bringup the k8s sample app", func() {
+			// Export check
+			assert.Equal(t, strings.Trim(riofile.Readfile(), "\n"), strings.Trim(riofile.ExportStack(), "\n"), "should have stack export be same as original file")
+
+			// check frontend service came up
+			guestBook := testutil.CreateRoute(t, "guestbook", "frontend", "to")
+			defer guestBook.Remove()
+			guestBook.GetEndpointResponse()
+			//nginix service
+			nginxsvc := testutil.GetService(t, "nginx", "v0")
+			nginxsvc.GetEndpointResponse()
+
+			//assert k8s services came up outside of riofile
+			riofile.Up(t, "riofile-with-K8s-manifests-after.yaml", riofileName)
+			defer riofile.Remove()
+
+			//check that services/pods/deployments are removed
+			time.Sleep(30 * time.Second)
```

Comment:
Is there a way that we can wait for the services/pods/deployments to be removed with a timeout instead of sleeping? Would hopefully fix potential flakiness on less powerful machines with tests running in parallel.

Reply:
+1, something like `GetService` for removals

Reply:
Updated to address this was a new function `WaitForNoResponse`